src/anomalib/models/image/vlm_ad/backends/huggingface.py

```diff
@@ -0,0 +1,99 @@
+"""Huggingface backend."""
+
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from pathlib import Path
+
+from PIL import Image
+from transformers.modeling_utils import PreTrainedModel
+
+from anomalib.models.image.vlm_ad.utils import Prompt
+from anomalib.utils.exceptions import try_import
+
+from .base import Backend
+
+if try_import("transformers"):
+    import transformers
+    from transformers.modeling_utils import PreTrainedModel
+    from transformers.processing_utils import ProcessorMixin
+else:
+    transformers = None
+
+
+logger = logging.getLogger(__name__)
+
+
+class Huggingface(Backend):
+    """Huggingface backend."""
+
+    def __init__(
+        self,
+        model_name: str,
+        api_key: str | None = None,
```

Comment:
Not sure if I understand the reason why `api_key` is listed as argument if it is not needed.